src/diagnostic/source_manager.rs

```diff
@@ -164,11 +164,19 @@ impl SourceManager {
         Ok(self.files.push_get(Box::new(Rc::new(source_file))).clone())
     }
 
-    /// Returns a reference to a SourceFile at the given index
+    /// Returns a reference to a SourceFile at the given index, or None if the index is invalid
     pub fn get_file(&self, index: usize) -> Option<Rc<SourceFile>> {
         self.files.get(index).cloned()
     }
 
+    /// Returns a reference to a SourceFile at the given index, or panics if the index is invalid
+    pub fn get_file_unwrap(&self, index: usize) -> Rc<SourceFile> {
+        self.files.get(index).cloned().expect(&format!(
+            "SourceManager recieved invalid SourceFile index ({}) during get_file_unwrap()",
+            index
+        ))
+    }
```

Comment:
This is kind of a rip (which clippy nicely picks up on) because the string has has to be formatted regardless of weather or not the `Option` contains `Some(...)`

Reply:
Fixed also in the latest commit a2db4aa68506fd60efda1b110015256e5de56139